src/allotropy/parsers/agilent_gen5/agilent_gen5_structure.py

```diff
@@ -1,26 +1,574 @@
+                for excitation, emission in zip(excitations, emissions)
+            ]
+
+        if read_mode == ReadMode.LUMINESCENCE:
+            emissions = device_control_data.get(EMISSION_KEY)
+            for emission in emissions:
+                label = "Lum" if emission == "Full light" else emission
+                measurement_labels.append(f"{label_prefix}{label}")
 
-        if not plates:
-            msg = "No plate data found in file."
+        return measurement_labels
+
+    @classmethod
+    def _get_absorbance_measurement_labels(
+        cls, label_prefix: Optional[str], device_control_data: dict
+    ) -> list:
+        wavelengths = device_control_data.get(WAVELENGTHS_KEY, [])
+        pathlength_correction = device_control_data.get(PATHLENGTH_CORRECTION_KEY)
+        measurement_labels = []
+
+        for wavelenght in wavelengths:
+            label = f"{label_prefix}{wavelenght}"
+            measurement_labels.append(label)
+
+        if pathlength_correction:
+            test, ref = pathlength_correction.split(" / ")
+            test_label = f"{label_prefix}{test} [Test]"
+            ref_label = f"{label_prefix}{ref} [Ref]"
+            measurement_labels.extend([test_label, ref_label])
+
+        return measurement_labels
+
+    @classmethod
+    def _get_device_control_data(
+        cls, procedure_details: str, read_mode: ReadMode
+    ) -> dict:
+        list_keys = frozenset(
+            {
+                EMISSION_KEY,
+                EXCITATION_KEY,
+                OPTICS_KEY,
+                GAIN_KEY,
+                MIRROR_KEY,
+                WAVELENGTHS_KEY,
+            }
+        )
+        read_data_dict: dict = {label: [] for label in list_keys}
+        read_lines: list[str] = procedure_details.splitlines()
+        datum_len = 2
+
+        for line in read_lines:
+            strp_line = str(line.strip())
+            if strp_line.startswith("Read\t"):
+                read_data_dict["Step Label"] = cls._get_step_label(line, read_mode)
+                continue
+
+            elif strp_line.startswith(WAVELENGTHS_KEY):
+                wavelengths = strp_line.split(":  ")
+                read_data_dict[WAVELENGTHS_KEY].extend(wavelengths[1].split(", "))
+                continue
+
+            line_data: list[str] = strp_line.split(",  ")
+            for read_datum in line_data:
+                splitted_datum = read_datum.split(": ")
+                if len(splitted_datum) != datum_len:
+                    continue
+                if splitted_datum[0] in list_keys:
+                    read_data_dict[splitted_datum[0]].append(splitted_datum[1])
+                else:
+                    read_data_dict[splitted_datum[0]] = splitted_datum[1]
+
+        return read_data_dict
+
+    @classmethod
+    def _get_step_label(cls, read_line: str, read_mode: str) -> Optional[str]:
+        read_data_len = 2
+        split_line = read_line.split("\t")
+        if len(split_line) != read_data_len:
```

Comment:
```suggestion
        if len(split_line) != 2:
```
Same comment as above.